src/components/LocaleSelector.vue

```diff
@@ -45,10 +46,11 @@ export default {
   },
   methods: {
     updateRouter({ target: { value: currentLocale } }) {
+      const slug = getSlug(currentLocale);
```

Comment:
To minimize the amount of necessary code/logic, I think you might actually want to populate the selector items with slug values. Since you're going to need to convert slugs to locale codes in the router guards (`updateCurrentLocale`) _anyway_, it would save you from also doing the opposite conversion here.

Essentially, you would be doing the reverse of this, ie `getCode(localeSlug)`—however, you wouldn't then need to add the reverse mapping in the router guard (which is missing here right now since it assumes the slugs are always the same).